mos_tests/neutron/python_tests/base.py

```diff
@@ -187,6 +187,36 @@ def check_vm_connectivity(self, timeout=3 * 60):
             self.check_ping_from_vm(server1, self.instance_keypair,
                                     ips_to_ping, timeout=timeout)
 
+    def run_on_cirros(self, vm, cmd):
+        """Run command on Cirros VM, connected by floating ip.
+
+        :param vm: instance with cirros
+        :param cmd: command to execute
+        :returns: dict, result of command with code, stdout, stderr.
+        """
+        vm = self.os_conn.get_instance_detail(vm)
+        _floating_ip = self.os_conn.get_nova_instance_ips(vm)['floating']
+
+        with self.env.get_ssh_to_vm(_floating_ip,
```

Comment:
It's code from ban dhcp agent and in some cases we need to direct connect by floating ip.